src/glob.rs

```diff
@@ -0,0 +1,68 @@
+//! Wrapper around globset's Glob type that has better serialization
+//! characteristics by coupling Glob and GlobMatcher into a single type.
+
+use std::{fmt, path::Path};
+
+use globset::{Glob as InnerGlob, GlobMatcher};
+use serde::{
+    de::{self, Visitor},
+    Deserialize, Deserializer, Serialize, Serializer,
+};
+
+pub use globset::Error;
+
+#[derive(Debug, Clone)]
+pub struct Glob {
+    inner: InnerGlob,
+    matcher: GlobMatcher,
+}
+
+impl Glob {
+    pub fn new(glob: &str) -> Result<Self, Error> {
+        let inner = InnerGlob::new(glob)?;
+        let matcher = inner.compile_matcher();
+
+        Ok(Glob { inner, matcher })
+    }
+
+    pub fn is_match<P: AsRef<Path>>(&self, path: P) -> bool {
+        self.matcher.is_match(path)
+    }
+}
+
+impl PartialEq for Glob {
+    fn eq(&self, other: &Self) -> bool {
+        self.inner == other.inner
+    }
+}
+
+impl Eq for Glob {}
+
+impl Serialize for Glob {
+    fn serialize<S: Serializer>(&self, serializer: S) -> Result<S::Ok, S::Error> {
+        serializer.serialize_str(self.inner.glob())
+    }
+}
+
+impl<'de> Deserialize<'de> for Glob {
+    fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
+        deserializer.deserialize_str(GlobVisitor)
```

Comment:
You can probably replace this with:

```rs
Glob::new(String::deserialize(deserializer))
```

... and rip out the visitor completely.

Reply:
Because `Glob::new` (both ours and the one from the globset crate) accept `&str` and not `String`, I think I'd rather keep the possibility that we can avoid copying the glob string here.

The visitor implementation isn't _so_ bad! 😄

Reply:
str::deserialize or `<&str as Deserialize>::deserialize` probably work too then if you still want to try it ☺️

Reply:
It ended up being a little weirder, but worked out!

```rust
let glob = <&str as Deserialize>::deserialize(deserializer)?;
Glob::new(glob).map_err(D::Error::custom)
```

Ended up having to `use serde::de::Error as _` as well to bring the `custom` method into scope.

Reply:
I'm not confident one way or the other, other than to say I do things like this without issue.